Add `run_with_db_tasks` config key

agrberg opened the pull request:
This will turn off the automatically run secondbase:db tasks

hmadison commented on README.md:
```diff
@@ -52,6 +52,8 @@ This will not only create your base development database, but it will also creat
 * db:test:prepare
 
+**These can all be deactivated by setting `config.second_base.run_with_db_rake_tasks = false` in your Application's configuration.**
+
```
I think this would make sense like:

> SecondBase aims to work seamlessly within your Rails application. When it makes sense, we run a mirrored db:second_base task for matching ActiveRecord base database task.These can all be deactivated by setting `config.second_base.run_with_db_rake_tasks = false` in your Application's configuration.

hmadison commented on lib/second_base/databases.rake:
```diff
@@ -113,7 +113,7 @@ end
 }.each do |name|
   task = Rake::Task["db:#{name}"] rescue nil
-  next unless task
+  next unless task && SecondBase::Railtie.run_with_db_rake_tasks?
```
`SecondBase::Railtie.run_with_db_rake_tasks?` needs to be mirrored in the rails 4 and rails 5 tasks too.

agrberg replied:
👍  thanks, I didn't know about those due to my yet unfamiliarity with the project. I'll have an amended PR up shortly.

hmadison commented:
Overall great idea. I made few small comments. If you need help getting the tests folded in I can add them this weekend and merge it.

agrberg commented:
Addressed the comments. Thanks for responding so quickly. I'd be very interested to see how you put the tests and how it differs from my stuck approach.

hmadison commented:
@agrberg I think this is what you want for the tests.

```
diff --git a/test/cases/dbtask_test.rb b/test/cases/dbtask_test.rb
index f911e2d..df87c7a 100644
--- a/test/cases/dbtask_test.rb
+++ b/test/cases/dbtask_test.rb
@@ -207,9 +207,19 @@ class DbTaskTest < SecondBase::TestCase
     assert_match(/version: 20151202075826/, run_secondbase(:version))
   end
 
+  def test_no_db_tasks
+    refute_dummy_databases
+    run_db :create, :stdin, false
+    assert_dummy_created_but_not_secondbase
+  end
 
   private
 
+  def assert_dummy_created_but_not_secondbase
+    assert_equal 'base.sqlite3', dummy_database_sqlite
+    refute_match(/secondbase_test/, `mysql -uroot -e "SHOW DATABASES"`)
+  end
+
   def assert_no_tables
     if ActiveRecord::Base.connection.respond_to? :data_sources
       assert_equal [], ActiveRecord::Base.connection.data_sources
diff --git a/test/dummy_apps/rails_five/init.rb b/test/dummy_apps/rails_five/init.rb
index f65de0a..1e8a826 100644
--- a/test/dummy_apps/rails_five/init.rb
+++ b/test/dummy_apps/rails_five/init.rb
@@ -33,6 +33,7 @@ module Dummy
 
     config.active_record.schema_format = ENV['SCHEMA_FORMAT'] ? :sql : :ruby
 
+    config.second_base.run_with_db_rake_tasks = ENV['WITH_SECONDBASE_TASKS'] == 'true'
 
   end
 end
diff --git a/test/dummy_apps/rails_four/init.rb b/test/dummy_apps/rails_four/init.rb
index f65de0a..1e8a826 100644
--- a/test/dummy_apps/rails_four/init.rb
+++ b/test/dummy_apps/rails_four/init.rb
@@ -33,6 +33,7 @@ module Dummy
 
     config.active_record.schema_format = ENV['SCHEMA_FORMAT'] ? :sql : :ruby
 
+    config.second_base.run_with_db_rake_tasks = ENV['WITH_SECONDBASE_TASKS'] == 'true'
 
   end
 end
diff --git a/test/test_helpers/dummy_app_helpers.rb b/test/test_helpers/dummy_app_helpers.rb
index 0633a92..122d82e 100644
--- a/test/test_helpers/dummy_app_helpers.rb
+++ b/test/test_helpers/dummy_app_helpers.rb
@@ -66,9 +66,9 @@ module SecondBase
       'rake'
     end
 
-    def run_db(args, stream=:stdout)
+    def run_db(args, stream=:stdout, with_secondbase_tasks=true)
       capture(stream) do
-        Dir.chdir(dummy_root) { Kernel.system "#{run_cmd} db:#{args}" }
+        Dir.chdir(dummy_root) { Kernel.system "env WITH_SECONDBASE_TASKS=#{with_secondbase_tasks} #{run_cmd} db:#{args}" }
       end
     end
 ```

metaskills commented:
Can we rename this config to something that does not include the `rake` name? Now that we support Rails 5, many tasks primary interface is with the `rails` bin. May I suggest something like `run_with_db_tasks` instead.

agrberg commented:
Updated key name and added tests (had to mess with them a little bit). I can get `bundle exec appraisal rails41 rake test` to pass but I'm still having problems with `bundle exec appraisal rake test` so I'll see what Travis says about the rest.

hmadison commented:
Looks good to me. @metaskills any last thoughts?

metaskills commented:
Nope, thanks for asking. 🎩

hmadison commented:
Released 2.1.0 with these changes. Thanks @agrberg!

agrberg commented:
Happy to help and thank you guys for helping me with the PR. 👍

karledurante commented:
@hmadison do you want to update the Change Log as well?
